src/filter/s0rter/s0rter.c

```diff
@@ -0,0 +1,255 @@
+#include "s0rter.h"
+
+// #define S0RTER_DEBUG 1;
+
+int f0r_init() {
+  return 1;
+}
+
+void f0r_deinit() {
+  // intentionally empty
+}
+
+void f0r_get_plugin_info(f0r_plugin_info_t* info) {
+  info->name = "s0rter";
+  info->author = "Sean Barag";
+  info->plugin_type = F0R_PLUGIN_TYPE_FILTER;
+  info->color_model = F0R_COLOR_MODEL_RGBA8888;
+  info->frei0r_version = FREI0R_MAJOR_VERSION;
+  info->major_version = 1;
+  info->minor_version = 0;
+  info->num_params = 0;
+  info->explanation = "Sorts the pixels in each frame by (h, s, l) tuple.";
+}
+
+void f0r_get_param_info(f0r_param_info_t* info, int param_index) {
+  // intentionally empty, but reference params because of -Wextra
+  (void)info;
+  (void)param_index;
+}
+
+void f0r_set_param_value(f0r_instance_t instance,
+                         f0r_param_t param,
+                         int param_index) {
+  // intentionally empty, but reference params because of -Wextra
+  (void)instance;
+  (void)param;
+  (void)param_index;
+}
+
+void f0r_get_param_value(f0r_instance_t instance,
+                         f0r_param_t param,
+                         int param_index) {
+  // intentionally empty, but reference params because of -Wextra
+  (void)instance;
+  (void)param;
+  (void)param_index;
+}
+
+f0r_instance_t f0r_construct(unsigned int width, unsigned int height) {
+  s0rter_instance_t* instance =
+      (s0rter_instance_t*)calloc(1, sizeof(*instance));
+  instance->width = width;
+  instance->height = height;
+  instance->hsl_buffer = (hsl_t*)calloc(width * height, sizeof(hsl_t));
```

Comment:
do you mean that some hosts use the same plugin instance in multiple concurrent threads? If I understood it well, this is rather odd: plugin instantiation was thought to allow safe multi-threading by creating a new instance per thread.

Reply:
I was thinking historically about MLT's usage. We do have a separate plugin instance per thread since Oct 2019, but it is a suggestion in case there is a host that does not. Something to consider as well is that each thread could have this large buffer for an extended period of time especially for long held plugin instances as is typically the case in a video editor thereby increasing the memory footprint of the process.